app/code/core/Mage/Sales/Model/Order.php

```diff
@@ -841,7 +844,10 @@ public function canEdit()
             return false;
         }
 
-        if (!$this->getPayment()->getMethodInstance()->canEdit()) {
+        if (!($payment = $this->getPayment())
+            || Mage::helper('payment')->getMethodModelClassName($payment->getMethod()) === null
+            || !$payment->getMethodInstance()->canEdit()
+        ) {
```

Comment:
here too i'd prefer multiple simpler ifs instead of a big difficult to read one

Reply:
No problem. I simplified this part so the code should be easier to read. :)